### PR TITLE
set auth in cluster context so it works with EKS

### PR DESCRIPTION
**Pull request by jensendw**
In order to make this resource work with a vanilla EKS setup I needed the following items to be supported.

### CODE REVIEW

**Comment by superbrothers** [comment]
`certificate_authority_data` looks like the same with `certificate_authority`.

```yaml
resources:
- name: kubernetes-production
  type: kubernetes
  source:
    server: https://abc123.xyz.us-west-2.eks.amazonaws.com
    certificate_authority: ((eks-o1_cluster.ca))
    use_aws_iam_authenticator: true
    aws_eks_cluster_name: my-eks-cluster
- name: my-app
  type: git
  source:
    ...
```

**Comment by jensendw** [comment]
@superbrothers it does indeed, unfortunately thats not the case.  When ran using that you will receive the following error 

```
+ kubectl version
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.2", GitCommit:"17c77c7898218073f14c8d573582e8d2313dc740", GitTreeState:"clean", BuildDate:"2018-10-24T06:54:59Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"linux/amd64"}
Unable to connect to the server: x509: certificate signed by unknown authority
```

For some reason I thought it would be a good idea to set insecure_skip_tls_verify: true, which then results in:
```
error: you cannot specify a certificate authority and insecure mode at the same time
```

For whatever reason the EKS documentation specifies that certificate-authority-data needs to be set for the CA cert as specified here https://docs.aws.amazon.com/eks/latest/userguide/create-kubeconfig.html

I have no idea what the difference is, and could not find anything in the kubectl documentation.  However, even when I test locally with a base setup of EKS it will not work if I specify certificate-authority, but will work if I specify certificate-authority-data.

**Comment by superbrothers** [comment]
Did you set a raw ca-cert (non-base64 encoded) to `certificate-authority` source param?

- `certificate_authority`: *Optional.* A certificate file for the certificate authority.
    ```yaml
    certificate_authority: |
        -----BEGIN CERTIFICATE-----
        ...
        -----END CERTIFICATE-----
    ```

**Comment by jensendw** [comment]
It never even occurred to me that the CA for certificate_authority shouldn't be base64 encoded.  So after fixing that here is where I'm at: 

Version 1.12: 
It errors with 
```ERROR: You must specify "server" and "token", if not specify "kubeconfig".```
This is presumably because the 1.12 doesn't reflect whats in master

My own built image from master:
```
kubectl apply -f git-kubernetes-cluster-config-eks-o1/eks-o1
Error from server (Forbidden): error when retrieving current configuration of:
Resource: "rbac.authorization.k8s.io/v1, Resource=clusterrolebindings", GroupVersionKind: "rbac.authorization.k8s.io/v1, Kind=ClusterRoleBinding"
```
If I hijack the task and look at the temp file I see this 
```
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: 
xxxxxxx
    server: https://xxxxx.us-west-2.eks.amazonaws.com
  name: cluster
contexts:
- context:
    cluster: cluster
    namespace: default
    user: ""
  name: kubernetes-resource
current-context: kubernetes-resource
kind: Config
preferences: {}
users:
- name: auth
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1alpha1
      args:
      - token
      - -i
      - eks-o1
      command: aws-iam-authenticator
      env: null
```
Notice how under context user is "", if I change that field to be auth it will work.  I will revise my PR to reflect just that update.  Thanks for pointing out the CA thing!

File: assets/common.sh [hunk 1]
```diff
@@ -63,7 +63,7 @@ setup_kubectl() {
     fi
     exe kubectl config set-cluster "$CLUSTER_NAME" "${set_cluster_opts[@]}"
 
-    exe kubectl config set-context "$CONTEXT_NAME" --cluster="$CLUSTER_NAME"
+    exe kubectl config set-context "$CONTEXT_NAME" --cluster="$CLUSTER_NAME" --user "$AUTH_NAME"
```
Comment (jensendw) [inline assets/common.sh]: With this fix the generated kubectl config file looks like this, when executed with a build of a container that has this code everything works as expected.
```
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: 
xxxxxxx
    server: https://xxxxx.us-west-2.eks.amazonaws.com
  name: cluster
contexts:
- context:
    cluster: cluster
    namespace: default
    user: auth
  name: kubernetes-resource
current-context: kubernetes-resource
kind: Config
preferences: {}
users:
- name: auth
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1alpha1
      args:
      - token
      - -i
      - eks-o1
      command: aws-iam-authenticator
      env: null
```

**Comment by jensendw** [comment]
@superbrothers Also, any info on rebuilding the 1.12 or latest docker image so it will work?  I realize your versioning is following the kubectl version, which makes sense, is your plan to rebuild the images with bug fixes or increment the docker tag version IE, 1.12.1?

**Comment by jensendw** [comment]
Any idea when this will get merged and a new release will be built?

**Comment by superbrothers** [comment]
@jensendw We came back from KubeCon China. I'm sorry for late response.

**Pull request merged**